### ISSUE TITLE
Seems to not work for big programs (lots of copybooks?)

### ISSUE THREAD

**Issue by GitMensch** [eclipse/che-che4z-lsp-for-cobol#1045]
## Description ##
Opening a COBOL workspace with production code I see that the java process consumes ~ 680MB (I have no idea if this is normal or not, my local sample workspace starts a java process  that consumes less than 120MB), I see the following output from the LSP

> Jul 28, 2021 2:50:43 PM org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint notify
INFORMATION: Unsupported notification method: $/setTraceNotification
Jul 28, 2021 2:50:44 PM org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint notify
INFORMATION: Unsupported notification method: $/setTraceNotification

... and no issue reported in the problem pane at all (even the "missing copybook" message when writing `COPY BAD.` is not created).

So I do wonder - is there something I can do to:

* possibly decrease the amount of RAM used (this is a shared machine and if every user wants > 600MB for every instance of vscode...)
* find out why there is no parsing of the file

?

**Comment by temanbrcom** [comment]
Hi @GitMensch,

I am sorry for the late response. 
The messages belong to the VSC and don't show any meaningful information for the extension.
Memory consumption is a real problem, now we are working hard to fix it, so you may expect improvements in the next release.
I cannot say when we release, I will keep you updated. Please, check then if the issue still presents and let me know. 

Kind regards, 
Andrei Temnikov.

**Comment by GitMensch** [comment]
Sure, will do, especially if you drop a hint then (otherwise I may miss that update because this extension isn't yet usable for me and therefore not used daily). I guess a merge of #1055 would already be enough for a new release, wouldn't it?

**Comment by temanbrcom** [comment]
Thank you for your enthusiasm!
We are waiting for #1055 and #1059 to start the release, I would say the next version will be available in a week. I will ping you when we are done.

**Comment by temanbrcom** [comment]
HI @GitMensch,

Thank you for your patience! We just released version 0.21.0 with multiple bug fixes and performance improvements. Please, check if it works for you.

Kind regards,
Andrei Temnikov

**Comment by GitMensch** [comment]
I wanted to check, but it isn't available on open-vsx yet.
https://open-vsx.org/extension/BroadcomMFD/cobol-language-support/changes

**Comment by GitMensch** [comment]
Rechecked from vsix - the dependency in 0.19.0 increased to vscode 1.56 - do you actually use one of the new features or did the version increase slipped in somehow? I'd check that myself - but without the source repo...

**Comment by temanbrcom** [comment]
I'm sorry, we have trouble deploying it there. I will let you know once we deal with it. 

Kind regards,
Andrei Temnikov

**Comment by temanbrcom** [comment]
We updated the vscode because it is our requirement to use the latest stable versions of the dependencies to maintain a high level of security and reliability.

**Comment by GitMensch** [comment]
> Thank you for your patience! We just released version 0.21.0 with multiple bug fixes and performance improvements. Please, check if it works for you.

@temanbrcom Yes, that works in general (tested with a clean portable vscodium and the existing big workspaces), during analysis I get a reasonable note "Syntax analysis in progress". After opening 4 files I got a vscodium executable with 272MB+91MB RAM usage and a java server process with 892 MB, so I *think* that's still too much.

Should we create a new issue for that and close this one?

**Comment by GitMensch** [comment]
![VisualVM Object view](https://user-images.githubusercontent.com/6699539/132877108-4d83a0a6-1269-441b-be50-c0b26db99e1c.png)

**Comment by GitMensch** [comment]
Other than the memory issue it seems I need to wait on #1114 and #1046 to be able to test it in these workspaces more.

**Comment by temanbrcom** [comment]
Well, the allocated memory strictly depends on the number of files and their size. We are working on decreasing memory usage, but I cannot promise any drastic changes.

**Comment by temanbrcom** [comment]
> Other than the memory issue it seems I need to wait on #1114 and #1046 to be able to test it in these workspaces more.

Sounds reasonable, we will discuss raising the priority of those issues.

**Comment by GitMensch** [comment]
> Well, the allocated memory strictly depends on the number of files and their size.

As I'm not "in the code" yet: Are all files in the workspace opened? Only those that are currently open in an editor (or even: only in active editors) along with the copybooks?

Note: I was able to reduce the memory for https://github.com/eclipse/lemminx, sometimes one just needs an additional eye... I may give that a try as the code is available and https://github.com/eclipse/che-che4z-lsp-for-cobol/blob/development/CONTRIBUTING.md looks reasonable (but I guess it will be quite some time before I can inspect that).

I'll close this one and create a new for the memory usage.

**Comment by GitMensch** [comment]
> 
> 
> We updated the vscode because it is our requirement to use the latest stable versions of the dependencies to maintain a high level of security and reliability.

Note: just checked: after manually downloading the vsix, opening it with 7-zip and editing both the manifest and the product.json - the current vsix also installs and seems to work well with vscode 1.55 :-)

**Comment by temanbrcom** [comment]
> > Well, the allocated memory strictly depends on the number of files and their size.
> 
> As I'm not "in the code" yet: Are all files in the workspace opened? Only those that are currently open in an editor (or even: only in active editors) along with the copybooks?
> 
> Note: I was able to reduce the memory for https://github.com/eclipse/lemminx, sometimes one just needs an additional eye... I may give that a try as the code is available and https://github.com/eclipse/che-che4z-lsp-for-cobol/blob/development/CONTRIBUTING.md looks reasonable (but I guess it will be quite some time before I can inspect that).
> 
> I'll close this one and create a new for the memory usage.

It would be really great if you do it! Please, feel free to check the code and ask questions in our [Slack channel](https://join.slack.com/t/che4z/shared_invite/zt-vlw321sg-2HuUaU0zI7fGIo2q7sEKZg).

And about the files - the server processes only the open files with the referenced copybooks, and they are disposed only when the tab is closed.

Kind regards,
Andrei Temnikov
